mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -226,55 +250,94 @@ impl State for WaitForTakerPaymentSpend {
     type Result = ();
 
     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherContext) -> StateResult<Self::Ctx, Self::Result> {
-        if std::env::var("SWAP_WATCHER_SKIP_WAITING").is_err() {
-            // Sleep for half the locktime to allow the taker to spend the maker payment first
-            let now = now_ms() / 1000;
-            let wait_for_taker_until =
-                wait_for_taker_payment_conf_until(watcher_ctx.data.swap_started_at, watcher_ctx.data.lock_duration);
-            let sleep_duration = (wait_for_taker_until - now + 1) as f64;
-
-            if now < wait_for_taker_until {
-                Timer::sleep(sleep_duration).await;
-            }
-        }
-
-        let f = watcher_ctx.taker_coin.wait_for_htlc_tx_spend(
-            &watcher_ctx.data.taker_payment_hex,
-            &[],
-            watcher_ctx.data.taker_payment_lock,
-            watcher_ctx.data.taker_coin_start_block,
-            &None,
-        );
-
-        let tx = match f.compat().await {
-            Ok(t) => t,
-            Err(err) => {
-                error!("{}", err.get_plain_text_format());
-                return Self::change_state(RefundTakerPayment {});
-            },
+        let payment_search_interval = match std::env::var("USE_TEST_SEARCH_INTERVAL") {
+            Ok(_) => TAKER_PAYMENT_SPEND_SEARCH_INTERVAL,
+            Err(_) => 300.,
```

Comment:
`const` value can be nice for `300.` value just like `TAKER_PAYMENT_SPEND_SEARCH_INTERVAL`